exp/internal/controllers/machinepool_controller.go

```diff
@@ -282,12 +291,26 @@ func (r *MachinePoolReconciler) reconcileDeleteNodes(ctx context.Context, cluste
 	return r.deleteRetiredNodes(ctx, clusterClient, machinepool.Status.NodeRefs, machinepool.Spec.ProviderIDList)
 }
 
+// isMachinePoolDeleteTimeoutPassed check the machinePool node delete time out.
+func (r *MachinePoolReconciler) isMachinePoolNodesDeleteTimeoutPassed(machinepool *expv1.MachinePool) bool {
```

Comment:
nit: given this operate against the MachinePoolReconciler and receives a machinePool, naming seems redundant i.e isNodesDeleteTimeoutPassed should be sufficient.

Reply:
The name originally was isMachinePoolDeleteTimeoutPassed based in your previous comment I made it isMachinePoolNodesDeleteTimeoutPassed. I believe the right name  isMachinePoolDeleteTimeoutPassed

Reply:
>The name originally was isMachinePoolDeleteTimeoutPassed based in your previous comment I made it isMachinePoolNodesDeleteTimeoutPassed.

Yes, thanks for bearing with me. In this iteration becomes clear the redundancy. Not a blocker to me if you want to keep it as it is.

> isMachinePoolDeleteTimeoutPassed

This is not a machinePool delete timeout. This is a timeout for Nodes deletion, which is a fraction of the machinePool deletion process.